mlrun/model.py

```diff
@@ -1514,26 +1533,49 @@ def ui_url(self) -> str:
 
     @property
     def outputs(self):
-        """return a dict of outputs, result values and artifact uris"""
-        outputs = {}
+        """
+        Return a dictionary of outputs, including result values and artifact URIs.
+
+        This method waits for the outputs to complete and combines result values
+        and artifact URIs into a single dictionary. If artifacts without URIs are
+        found, their targets are fetched and added to the dictionary.
+
+        :return: Dictionary containing result values and artifact URIs.
+        """
         self._outputs_wait_for_completion()
+        outputs = {}
+
+        # Add results if available
         if self.status.results:
-            outputs = {k: v for k, v in self.status.results.items()}
-        if self.status.artifacts:
-            for a in self.status.artifacts:
-                key = a["metadata"]["key"]
-                outputs[key] = get_artifact_target(a, self.metadata.project)
+            outputs.update(self.status.results)
+
+        # Add artifact URIs if available
+        if self.status.artifact_uris:
+            outputs.update(self.status.artifact_uris)
+
+        # Add artifacts with targets if available
+        elif self.status.artifacts:
+            for artifact in self.status.artifacts:
+                key = artifact["metadata"]["key"]
+                outputs[key] = get_artifact_target(artifact, self.metadata.project)
```

Comment:
same here. you need to get the artifact target path per key (it might be duplicated with the artifact uris)